src/main/java/com/iota/iri/service/validation/impl/TransactionSolidifierImpl.java

```diff
@@ -0,0 +1,389 @@
+        if (!transactionRequester.isTransactionRequested(hashPointer)) {
+            transactionRequester.requestTransaction(hashPointer);
+        }
+    }
+
+    /**
+     * Iterate through analyzed hashes and place them in the {@link #transactionsToBroadcast} queue
+     * @param hashes    Analyzed hashes from the {@link #checkSolidity(Hash)} call
+     */
+    private void updateTransactions(Set<Hash> hashes) {
+        hashes.forEach(hash -> {
+            try {
+                TransactionViewModel tvm = fromHash(tangle, hash);
+                tvm.updateHeights(tangle, snapshotProvider.getInitialSnapshot());
+
+                if(!tvm.isSolid()){
+                    tvm.updateSolid(true);
+                    tvm.update(tangle, snapshotProvider.getInitialSnapshot(), "solid|height");
+                }
+                addToBroadcastQueue(tvm);
+                addToPropagationQueue(tvm.getHash());
+            } catch (Exception e) {
+                log.info(e.getMessage());
+            }
+        });
+    }
+
+    /**
+     * Returns true if transaction is not solid and there are no solid entry points from the initial snapshot.
+     */
+    private boolean isUnsolidWithoutEntryPoint(TransactionViewModel transaction, Hash hashPointer) throws Exception{
+        if(!transaction.isSolid()){
+            if(!snapshotProvider.getInitialSnapshot().hasSolidEntryPoint(hashPointer)){
+                return true;
+            }
+        }
+        addToPropagationQueue(hashPointer);
+        return false;
+    }
+
+
+    private void addToBroadcastQueue(TransactionViewModel tvm) {
+        try {
+            if (transactionsToBroadcast.size() >= MAX_SIZE) {
+                transactionsToBroadcast.remove();
+            }
+
+            transactionsToBroadcast.put(tvm);
+        } catch(Exception e){
+            log.info("Error placing transaction into broadcast queue: " + e.getMessage());
+        }
+    }
+
+    @VisibleForTesting
+    Set<Hash> getSolidificationQueue(){
+        return new LinkedHashSet<>(transactionsToSolidify);
+    }
+
+
+    @Override
+    public void updateStatus(TransactionViewModel transactionViewModel) throws Exception {
+        transactionRequester.clearTransactionRequest(transactionViewModel.getHash());
+        if(transactionViewModel.getApprovers(tangle).size() == 0) {
+            tipsViewModel.addTipHash(transactionViewModel.getHash());
+        }
+        tipsViewModel.removeTipHash(transactionViewModel.getTrunkTransactionHash());
+        tipsViewModel.removeTipHash(transactionViewModel.getBranchTransactionHash());
+
+        if(quickSetSolid(transactionViewModel)) {
+            transactionViewModel.update(tangle, snapshotProvider.getInitialSnapshot(), "solid|height");
+            tipsViewModel.setSolid(transactionViewModel.getHash());
+            addToPropagationQueue(transactionViewModel.getHash());
+        }
+    }
+
+    public void addToPropagationQueue(Hash hash) throws Exception{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Public method and constructor comments are required](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=4339340)